xonsh/procs/pipelines.py

```diff
@@ -629,15 +629,30 @@ def inp(self):
         """Creates normalized input string from args."""
         return " ".join(self.args)
 
+    def get_formatted_lines(self, lines):
+        """Format output lines."""
+        format = XSH.env.get("XONSH_SUBPROC_OUTPUT_FORMAT", "stream_lines")
+        if format == "stream_lines":
+            return "".join(lines)
+        elif format == "list_lines":
+            if not lines:
+                return lines
+            elif len(lines) == 1:
+                return lines[0].rstrip("\n")
+            else:
+                return [line.rstrip("\n") for line in lines]
+        elif callable(format):
+            return format(lines)
+
```

Comment:
I think this is a good idea.

Additionally, should we do 
```python
            elif len(lines) == 1:
                return lines[0].rstrip("\n")
```

whether we're in `stream_lines` or `list_lines`?